Dockerfile

```diff
@@ -1,294 +1,550 @@
-                ;; \
-           esac
+    GOBIN=/out GO111MODULE=on go install "github.com/tc-hib/go-winres@${GOWINRES_VERSION}" \
+    && /out/go-winres --help
 
-FROM base AS swagger
-WORKDIR $GOPATH/src/github.com/go-swagger/go-swagger
-
-# GO_SWAGGER_COMMIT specifies the version of the go-swagger binary to build and
-# install. Go-swagger is used in CI for validating swagger.yaml in hack/validate/swagger-gen
-#
-# Currently uses a fork from https://github.com/kolyshkin/go-swagger/tree/golang-1.13-fix,
-# TODO: move to under moby/ or fix upstream go-swagger to work for us.
-ENV GO_SWAGGER_COMMIT c56166c036004ba7a3a321e5951ba472b9ae298c
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/go/pkg/mod \
-    --mount=type=tmpfs,target=/go/src/ \
-        set -x \
-        && git clone https://github.com/kolyshkin/go-swagger.git . \
-        && git checkout -q "$GO_SWAGGER_COMMIT" \
-        && go build -o /build/swagger github.com/go-swagger/go-swagger/cmd/swagger
+# containerd
+FROM base AS containerd-src
+WORKDIR /usr/src/containerd
+RUN git clone https://github.com/containerd/containerd.git .
+ARG CONTAINERD_VERSION
+RUN git fetch origin && git checkout -q "$CONTAINERD_VERSION"
 
-FROM debian:${BASE_DEBIAN_DISTRO} AS frozen-images
-ARG DEBIAN_FRONTEND
-RUN --mount=type=cache,sharing=locked,id=moby-frozen-images-aptlib,target=/var/lib/apt \
-    --mount=type=cache,sharing=locked,id=moby-frozen-images-aptcache,target=/var/cache/apt \
-       apt-get update && apt-get install -y --no-install-recommends \
-           ca-certificates \
-           curl \
-           jq
-# Get useful and necessary Hub images so we can "docker load" locally instead of pulling
-COPY contrib/download-frozen-image-v2.sh /
-ARG TARGETARCH
-RUN /download-frozen-image-v2.sh /build \
-        busybox:latest@sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629f637a209 \
-        busybox:glibc@sha256:1f81263701cddf6402afe9f33fca0266d9fff379e59b1748f33d3072da71ee85 \
-        debian:bullseye-slim@sha256:dacf278785a4daa9de07596ec739dbc07131e189942772210709c5c0777e8437 \
-        hello-world:latest@sha256:d58e752213a51785838f9eed2b7a498ffa1cb3aa7f946dda11af39286c3db9a9 \
-        arm32v7/hello-world:latest@sha256:50b8560ad574c779908da71f7ce370c0a2471c098d44d1c8f6b513c5a55eeeb1
-# See also frozenImages in "testutil/environment/protect.go" (which needs to be updated when adding images to this list)
+FROM base AS containerd-base-static
+WORKDIR /go/src/github.com/containerd/containerd
+ENV CGO_ENABLED=1
+ENV BUILDTAGS="netgo osusergo static_build"
+ENV EXTRA_LDFLAGS='-extldflags "-fno-PIC -static"'
+ENV GO111MODULE=off
 
-FROM base AS cross-false
+FROM base AS containerd-base-dynamic
+WORKDIR /go/src/github.com/containerd/containerd
+ENV GO111MODULE=off
 
-FROM --platform=linux/amd64 base AS cross-true
+FROM containerd-base-${DOCKER_LINKMODE} AS containerd-build
```

Comment:
Sure we can do that too but I think it's more elegant and readable that way.

Reply:
Hm, right, but now (I _think_) we're building / installing the shared bits _twice_, and I have to "scroll up" to learn what environment variables are set (which are used as part of the build / compile).

I think there's value in keeping those environment variables local / close to where they're used, so that I can see at a glance what's used 🤔

Reply:
Fair enough I will move them in `RUN` if you prefer :+1: